vulcanai2/models/basenetwork.py

```diff
@@ -489,6 +492,7 @@ def _init_criterion(criterion_spec):
         return criterion_spec
 
     def _init_trainer(self):
+        self._net_spec_initialzed = True
```

Comment:
If you want to determine if the optimizer was build you can simply do `if self.optim` and as long as it's None it's there. We don't need a class attribute just for this